sda/cmd/api/api.go

```diff
@@ -285,6 +292,61 @@ func ingestFile(c *gin.Context) {
 	c.Status(http.StatusOK)
 }
 
+// The deleteFile function deletes files from the inbox and marks them as
+// discarded in the db. Files are identified by their ids and the user id.
+func deleteFile(c *gin.Context) {
+	inbox, err := storage.NewBackend(Conf.Inbox)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	submissionUser := c.Param("username")
+	log.Debug("submission user:", submissionUser)
+
+	fileID := c.Param("file")
+	fileID = strings.TrimPrefix(fileID, "/")
+	log.Debug("submission file:", fileID)
+	if fileID == "" {
+		c.AbortWithStatusJSON(http.StatusBadRequest, "file ID is required")
+	}
+
+	// Get the file path from the fileID and submission user
+	filePath, err := Conf.API.DB.GetInboxFilePathFromID(submissionUser, fileID)
+	if err != nil {
+		log.Errorf("getting file from fileID failed, reason: (%v)", err)
+		c.AbortWithStatusJSON(http.StatusNotFound, "File could not be found in inbox")
+
+		return
+	}
+
+	// Requires a filepath instead of fileID
+	var RetryTimes = 5
+	for count := 1; count <= RetryTimes; count++ {
+		log.Warn("trying to remove file from inbox, try", count)
+		err = inbox.RemoveFile(filePath)
+		if err != nil {
+			log.Errorf("Remove file from inbox failed, reason: %v", err)
+		}
+
+		// The GetFileSize returns zero in case of error after retrying a number of times
+		fileSize, _ := inbox.GetFileSize(filePath)
+		if fileSize == 0 {
+			break
+		}
+
+		time.Sleep(time.Duration(math.Pow(2, float64(count))) * time.Second)
+	}
```

Comment:
```suggestion
	for count := 1; count <= 5; count++ {
		err := inbox.RemoveFile(filePath)
		if err == nil {
			break
		}

		if count == 5 {
			log.Errorf("remove file from inbox failed, reason: %v", err)
			c.AbortWithStatusJSON(http.StatusInternalServerError, ("remove file from inbox failed")
			
			return
		}

		time.Sleep(time.Duration(math.Pow(2, float64(count))) * time.Second)
	}
```

Reply:
Fixed in ebce118.

Reply:
I have not changed this. I think it may be good for refactoring, outside of this PR?

Reply:
Now fixed in 2dd0e1b.